CRM/Contribute/BAO/Contribution.php

```diff
@@ -3826,18 +3836,6 @@ public static function updateFinancialAccounts(&$params, $context = NULL) {
     foreach ($params['line_item'] as $fieldId => $fields) {
       $params = self::createFinancialItemsForLine($params, $context, $fields, $previousLineItems, $inputParams, $isARefund, $trxnIds, $fieldId);
     }
-
-    if ($context == 'changeFinancialType') {
```

Comment:
We managed to get this out of the shared function - yay - because this is kinda weird & a bit scary it's better to have it only in the place it affects